src/main/java/frc/robot/commands/FireIntoAmp.java

```diff
@@ -12,54 +12,78 @@
 import edu.wpi.first.math.geometry.Rotation2d;
 import edu.wpi.first.math.geometry.Translation2d;
 import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants;
 import frc.robot.Constants.LiftConstants;
 import frc.robot.subsystems.Launcher;
 import frc.robot.subsystems.Lift;
+import frc.robot.subsystems.Vision;
 import frc.robot.util.FieldPose2024;
 
 public class FireIntoAmp extends Command {
   private Lift m_lift;
   private Launcher m_launcher;
   private BaseSwerveDrive m_swerveDrive;
+  private Vision m_vision;
   private ampState state = ampState.moveIntoPosition;
+  private Pose2d m_finalPose;
 
   /** Creates a new FireIntoAmp. */
-  public FireIntoAmp(Lift lift, Launcher launcher, BaseSwerveDrive swerveDrive) {
+  public FireIntoAmp(Lift lift, Launcher launcher, BaseSwerveDrive swerveDrive, Vision vision) {
     m_lift = lift;
     m_launcher = launcher;
     m_swerveDrive = swerveDrive;
+    m_vision = vision;
     addRequirements(lift, launcher, swerveDrive);
 
   }
 
   public enum ampState {
-    moveIntoPosition, fireIntoAmp
+    moveIntoPosition, vision, fireIntoAmp
   }
 
   // Called when the command is initially scheduled.
   @Override
   public void initialize() {
-    double x = 0.7 * Math.cos(FieldPose2024.Amp.getCurrentAlliancePose().getRotation().getRadians());
-    double y = 0.7 * Math.sin(FieldPose2024.Amp.getCurrentAlliancePose().getRotation().getRadians());
+    double x = Constants.robotLengthMeters/2 * Math.cos(FieldPose2024.Amp.getCurrentAlliancePose().getRotation().getRadians());
+    double y = Constants.robotLengthMeters/2 * Math.sin(FieldPose2024.Amp.getCurrentAlliancePose().getRotation().getRadians());
     Translation2d pose = FieldPose2024.Amp.getCurrentAlliancePose().getTranslation().plus(new Translation2d(x, y));
     m_swerveDrive.setTarget(new Pose2d(pose, FieldPose2024.Amp.getCurrentAlliancePose().getRotation()));
+    m_finalPose = new Pose2d(pose, FieldPose2024.Amp.getCurrentAlliancePose().getRotation());
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
```

Comment:
We might want to revisit this based on mentor conversations today.

Start in ampState.vision, and also move the lift and launcher adjustments into that state. We aren't going to trust swerve Odometry enough to guide the robot into the general area.

Additionally, need to move some of these values into either Constants.java or command constants.